tensorly/decomposition/tests/test_cp.py

```diff
@@ -25,7 +25,13 @@ def test_parafac(linesearch, orthogonalise, true_rank, rank, init, monkeypatch):
     tol_norm_2 = 0.01
     tol_max_abs = 0.05
     tensor = random_cp((6, 8, 4), rank=true_rank, orthogonal=orthogonalise, full=True, random_state=rng)
-    fac, errors = parafac(tensor, rank=rank, n_iter_max=200, init=init, tol=10e-5, random_state=rng, orthogonalise=orthogonalise, linesearch=linesearch, return_errors=True)
+
+    # Callback to record error
+    errors = list()
+    def callback(_, __, rec_error):
+        errors.append(rec_error)
+
+    fac = parafac(tensor, rank=rank, n_iter_max=200, init=init, tol=10e-5, random_state=rng, orthogonalise=orthogonalise, linesearch=linesearch, callback=callback)
```

Comment:
Should we add some assert here, such as 
```python
fac, rec_errors = parafac(tensor, rank=rank, n_iter_max=200, init=init, tol=10e-5, random_state=rng, orthogonalise=orthogonalise, linesearch=linesearch, return_errors=True, callback=callback)
assert errors == rec_errors
```

Reply:
I've opted to not add this assert as I expect we are getting rid of rec_errors (to ensure the method functions have a more consistent return structure).